src/main/java/uk/gov/hmcts/darts/automation/utils/JsonApi.java

```diff
@@ -0,0 +1,282 @@
+					.header("Authorization", authorization)
+					.baseUri(baseUri)
+					.basePath("")
+					.body(body)
+				.when()
+					.post(endpoint)
+				.then()
+					.spec(responseLogLevel(ReadProperties.responseLogLevel))
+					.assertThat().statusCode(201)
+					.extract().response();
+		return response;
+    }
+    
+    public Response putApi(String endpoint, String body) {
+
+    	authorization = authenticate();
+    	log.info("put: " + endpoint);
+		response = 
+				given()
+					.spec(requestLogLevel(ReadProperties.requestLogLevel))
+					.accept("application/json, text/plain, */*")
+					.header("User-Agent","Mozilla/5.0 (Macintosh; Intel Mac OS X 10.14; rv:89.0) Gecko/20100101 Firefox/89.0") 
+					.header("Accept", "application/json, text/plain, */*")
+					.header("Accept-Encoding", "gzip, deflate")
+					.header("Content-Type", "application/json")
+					.header("Connection", "keep-alive")
+					.header("Authorization", authorization)
+					.baseUri(baseUri)
+					.basePath("")
+					.body(body)
+				.when()
+					.put(endpoint)
+				.then()
+					.spec(responseLogLevel(ReadProperties.responseLogLevel))
+					.assertThat().statusCode(204)
+					.extract().response();
+		return response;
+    }
+    
+    public Response deleteApi(String endpoint, String body) {
+
+    	log.info("delete: " + endpoint);
+    	authorization = authenticate();
+		response = 
+				given()
+					.spec(requestLogLevel(ReadProperties.requestLogLevel))
+					.accept("application/json, text/plain, */*")
+					.header("User-Agent","Mozilla/5.0 (Macintosh; Intel Mac OS X 10.14; rv:89.0) Gecko/20100101 Firefox/89.0") 
+					.header("Accept", "application/json, text/plain, */*")
+					.header("Accept-Encoding", "gzip, deflate")
+					.header("Content-Type", "application/json")
+					.header("Connection", "keep-alive")
+					.header("Authorization", authorization)
+					.baseUri(baseUri)
+					.basePath("")
+				.when()
+					.delete(endpoint)
+				.then()
+					.spec(responseLogLevel(ReadProperties.responseLogLevel))
+					.assertThat().statusCode(204)
+					.extract().response();
+		return response;
+    }
+    
+
+@Test
+	public void test() {
+    	JsonApi jsonApi = new JsonApi();
+//		System.out.println("post: courthouses");
+//		api.postApi("courthouses", "{\n"
+//    			+ "  \"courthouse_name\": \"Newcastle\",\n"
+//    			+ "  \"code\": 400\n"
+//    			+ "}");
+		System.out.println("get: courthouses");
+		jsonApi.getApi("courthouses");
+//		System.out.println("put: courthouses");
+//		api.putApi("courthouses/3", "{\n"
+//    			+ "  \"courthouse_name\": \"Leeds\",\n"
+//    			+ "  \"code\": 100\n"
+//    			+ "}");
+//		System.out.println("get2: courthouses");
+//		api.getApi("courthouses");
+//		System.out.println("delete: courthouses");
+//		deleteApi("courthouses/2", "{\n"
+//    			+ "  \"courthouse_name\": \"Leeds\",\n"
+//    			+ "  \"code\": 100\n"
+//    			+ "}");
+//		System.out.println("get2: courthouses");
+//    	getApi("courthouses");
```

Comment:
Commented out code should be removed as git manages any history of changes. If its a temporary change then you should have a description explaining why the code is commented out.

Reply:
I am refactoring this class as part of DMP-847 & will do that there